### PR TITLE
Docs/Packaging guide: Add BundlePackage

### PR DESCRIPTION
**Pull request by tldahlgren**
This PR adds ``BundlePackage`` to the packaging guide and updates the corresponding build system page to reflect some of the features people are currently using in their BundlePackages.

### CODE REVIEW

**Comment by tldahlgren** [comment]
Weird that I'm getting project coverage error showing up in the checks when this is a docs-only change AND the link is indicating an increase in coverage.

**Comment by tldahlgren** [comment]
Coverage failure is confusing.  Will close and re-open to see if that helps _at all_.

File: lib/spack/docs/build_systems/bundlepackage.rst [hunk 1]
```diff
@@ -9,9 +9,20 @@
...
+If it makes sense, variants and conflicts can be added to the package.
+Variants could ensure that common build options are consistent across
+the packages supporting them. Conflicts are used to prevent builds with
+known bugs or issues. For example, if the bundle is known to only build
+on ``linux`` then conflicts can be added, such as:
+
+.. code-block:: python
+
+    for platform in ["cray", "darwin", "windows"]:
+        conflicts(f"platform={platform}", msg="Only builds on linux")
```
Comment (tgamblin) [inline lib/spack/docs/build_systems/bundlepackage.rst]: why not:

```python
    require("platform=linux")
```

also some links to bundle packages (like xSDK or the vis SDK) might be useful

File: lib/spack/docs/packaging_guide.rst [hunk 2]
```diff
@@ -610,7 +641,16 @@ add a line like this in the package class:
        version("8.2.0", md5="1c9f62f0778697a09d36121ead88e08e")
        version("8.1.2", md5="d47dd09ed7ae6e7fd6f9a816d7f5fdf6")
 
-Versions should be listed in descending order, from newest to oldest.
+.. note::
+
+   Versions should be listed in descending order, from newest to oldest.
```
Comment (tgamblin) [inline lib/spack/docs/packaging_guide.rst]:
```suggestion
   By convention, we list versions in descending order, from newest to oldest.
```

**Review by tgamblin — CHANGES_REQUESTED** [review]

File: lib/spack/docs/build_systems/bundlepackage.rst (hunk 1 above)
Comment (tldahlgren) [reply lib/spack/docs/build_systems/bundlepackage.rst]: I was trying to illustrate the use of the for-loop above the directive so maybe that isn't the best example.

Great point about the links.  Thanks.

Comment (tgamblin) [reply lib/spack/docs/build_systems/bundlepackage.rst]: Not sure that the for loop really matters for the point here (though you'll get it across if you just link to existing sdk's -- they have for loops).

If you want to show require and conflict, you could show require for linux, and maybe a `conflicts("platform=windows")` for something that only works on unix

**Review by tgamblin — DISMISSED** [review]

**Comment by tldahlgren** [comment]
@tgambln I checked the output and noticed I left `Variants` in place when I added the link (so duplicates).

**Comment by tldahlgren** [comment]
"Fun" CI failure (https://gitlab.spack.io/spack/spack/-/pipelines/489104/failures) :

```
You may need to use sudo with that command.  After installing the python3-venv
package, recreate your virtual environment.

Failing command: ['/builds/spack/spack/pre_build_venv/bin/python3', '-Im', 'ensurepip', '--upgrade', '--default-pip']
```

**Review by tgamblin — APPROVED** [review]

**Pull request merged**